src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+            return;
+        let nodes = (event.target as HTMLSlotElement).assignedElements();
+        let name = (event.target as HTMLSlotElement)?.name;
+        if(name == 'tab-body') {
+            for(let i of nodes)
+                if(!i.classList.contains(this.activecls))
+                    (i as HTMLElement).style.display = 'none';
+        } else if(name == 'acc') {
+            for(let i of nodes)
+                if(!i.children[1].classList.contains(this.activecls))
+                    (i.children[1] as HTMLElement).style.display = 'none';
+        }
+    }
+    private _openActive() {
+        let val = this.querySelectorAll("[slot=tab-title]");
+        for(let i = 0;i < val.length; i++) {
+            if(val[i].classList.contains(this.activecls)) {
+                this.show(i);
+                break;
+            }
+        }
+        if(!this._prev_ix)
+            this.show(0);
+    }
+    private _checkFashion():string {
+        let vw = window.innerWidth;
+        for(let style of this._fashion_style)
+            if((style.ll || 0) <= vw && (
+                    !style.ul || style.ul >= vw)) {
+                return style.val;
+            }
+        return '';
+    }
+    private _changeFashionStyle(val:string) {
+        if(this._internal_fashion != val) {
+            if(val == 'accordion')
+                this._showAccordion();
+            else
+                this._showRC();
+            this._internal_fashion = val;
+        }
+    }
+    private _resizeEvent() {
+        let val = this._checkFashion();
+        this._changeFashionStyle(val);
+        this._openActive();
+    }
+    private _showAccordion() {
+        if(this.animation) {
+            let style = document.createElement("style");
+            style.innerHTML = `${animationCSS(
+                '.animation-open', '.animation-close',
+                this._animation_styles, this.animationDur)}`
+            this.appendChild(style);
+        }
+        let title = this.querySelectorAll("[slot=tab-title]");
+        let body = this.querySelectorAll("[slot=tab-body]");
+        for (let i = 0;i < Math.min(title.length, body.length); i++) {
+            let div = document.createElement("div");
+            div.slot = "acc";
+            div.appendChild(title[i]);
+            div.appendChild(body[i]);
+            this.appendChild(div);
+        }
+    }
+    private _showRC() {
+        let acc = this.querySelectorAll("[slot=acc]");
+        for(let j of acc) {
+            this.appendChild(j.children[1]);
+            this.appendChild(j.children[0]);
+            j.remove();
+        }
+        let styles = this.querySelectorAll("style");
+        for (let s of styles)
+            s.remove();
+    }
+    private _titleClicked(e:Event) {
+        let title = this.querySelectorAll("[slot=tab-title]");
+        for(let ix = 0;ix < title.length; ix++)
+            if(title[ix] == e.target && this._prev_ix != ix)
+                this.show(ix);
```

Comment:
Instead of having this loop, better to iterate the target's parent till we hit an element with `el.slot==tab-title` and the find index of that element within the querySelectorAll and show that index.